instrumentation/redis/lib/opentelemetry/instrumentation/redis/patches/client.rb

```diff
@@ -10,36 +10,25 @@ module Redis
       module Patches
         # Module to prepend to Redis::Client for instrumentation
         module Client
-          def call(*args, &block)
-            response = nil
-
+          def process(commands)
+            reply = nil
             attributes = client_attributes
-            attributes['db.statement'] = Utils.format_statement(args)
+            attributes['db.statement'] = Utils.format_statements(commands)
             tracer.in_span(
-              Utils.format_command(args),
+              Utils.format_span_name(commands),
               attributes: attributes,
               kind: :client
-            ) do
-              response = super(*args, &block)
+            ) do |s|
+              reply = super(commands)
+              if reply.is_a?(::Redis::CommandError)
+                s&.record_exception(reply)
+                s&.status = Trace::Status.new(
+                  Trace::Status::ERROR,
+                  description: "Unhandled exception of type: #{reply.class}"
```

Comment:
I don't think we should call this "Unhandled" - any error handling would occur higher in the call chain. Using the `message` is probably more descriptive.
```suggestion
                  description: reply.message
```